### DISCUSSION TITLE
neutron star and standard MS star binaries

### DISCUSSION THREAD

**Discussion by MohammadReza0GH** [phoebe-project/phoebe2#853] [Questions]
hi, how can i build a close binary when one of the star is neutron star?? i try but i got problems with irrad effect, that must have  significant  effect on lc in those binaries but i see no difference when i do irrad_method ='none' and i dont.

**Comment by kecnry** [comment]
Can you show a script of the setup that you have and what errors it is giving you?  You'll likely need to choose the atmosphere tables carefully (perhaps even falling back on blackbody atmospheres) on at least the neutron star and perhaps the MS star if it is sufficiently irradiated that the local temperatures go out of range of the default ck2004 tables.

**Reply by MohammadReza0GH** [reply]
import phoebe
b = phoebe.default_binary()
b1 = phoebe.default_binary()
logger = phoebe.logger(clevel = 'WARNING')
import matplotlib.pyplot as plt

b.add_constraint('semidetached', 'primary')
b.set_value('period@binary', value = 0.1725)
b.set_value('incl@binary@orbit', value = 51.7)
b.set_value('teff', component = 'primary', value = 3790)
b.set_value('teff', component = 'secondary', value = 7000)
b.set_value('requiv', component = 'secondary', value = 0.000014)
b.set_value('sma', component = 'binary', value= 3.8)
b.set_value('q', value = 5.7)
b.set_value('atm', component='primary', value='blackbody')
b.set_value('atm', component='secondary', value='blackbody')
b.get_parameter('irrad_frac_refl_bol@primary').set_value(0)
b.get_parameter('irrad_frac_refl_bol@secondary').set_value(0)
b.set_value('eclipse_method', value ='only_horizon')
b1.add_constraint('semidetached', 'primary')
b1.set_value('period@binary', value = 0.1725)
b1.set_value('incl@binary@orbit', value = 51.7)
b1.set_value('teff', component = 'primary', value = 3790)
b1.set_value('teff', component = 'secondary', value = 7000)
b1.set_value('requiv', component = 'secondary', value = 0.000014)
b1.set_value('sma', component = 'binary', value= 3.8)
b1.set_value('q', value = 5.7)
b1.set_value('atm', component='primary', value='blackbody')
b1.set_value('atm', component='secondary', value='blackbody')
b1.set_value('irrad_method', value='wilson')
#b1.get_parameter('irrad_frac_refl_bol@primary').set_value(1)
b1.get_parameter('irrad_frac_refl_bol@secondary').set_value(1)
b1.set_value('eclipse_method', value ='only_horizon')
b.add_dataset('lc', compute_times=phoebe.linspace(0,0.2,101),ld_mode='manual',pblum_mode='absolute',dataset= 'lc01')
b1.add_dataset('lc', compute_times=phoebe.linspace(0,0.2,101),ld_mode='manual',pblum_mode='absolute',dataset='lc02')
#b.add_dataset('mesh', compute_times=phoebe.linspace(0,0.1725,101),ld_mode='manual',dataset = 'mesh01')
#b.set_value('columns', value = ['teffs','loggs','*intensities*'])
b.set_value('ld_mode_bol', component='primary', value='manual')
b.set_value('ld_mode_bol', component = 'secondary', value='manual')
b1.set_value('ld_mode_bol', component='primary', value='manual')
b1.set_value('ld_mode_bol', component = 'secondary', value='manual')
b.run_compute()
b1.run_compute()
#b.plot(animate = True, uncover = True, fc = 'abs_normal_intensities',draw_sidebars=True, ec = "face",show=True,save ='eee.gif', save_kwargs={'writer': 'pillow'}, )
a = b.get_value('fluxes@lc01@phoebe01@latest@lc@model')
a1 =sorted(b.get_value('times@lc01@phoebe01@latest@lc@model'))
a2 = b1.get_value('fluxes@lc02@phoebe01@latest@lc@model')
a3 =sorted(b1.get_value('times@lc02@phoebe01@latest@lc@model'))
plt.plot(a1,a, label = '1')
plt.plot(a3,a2, label = '2')
plt.legend()
plt.show(True)

**Reply by kecnry** [reply]
There seem to be other small changes between these two (`b` and `b1`) besides just the albedoes, but if I subtract the fluxes, the light curves _are_ different - just at the order of 10^13 compared to the fluxes on the order of 10^23.  I suspect the NS is just not "heating up" the MS star enough to make a noticeable difference in the light curve relative to the ellipsoidal variations of the MS which is quite large and contributing quite a bit of light.

```
>>> b.get_value('fluxes', context='model') - b1.get_value('fluxes', context='model')
array([-1.86830742e+13, ..., -1.38536183e+13])
>>> b.get_value('fluxes', context='model')
array([2.37237422e+23, ..., 2.72285232e+23])
```

Here is a plot of the temperature at quarter-phase:
<img width="722" alt="Screen Shot 2024-04-23 at 3 37 56 PM" src="https://github.com/phoebe-project/phoebe2/assets/877591/caf43a89-93b8-4267-b077-e62a638791eb">

**Reply by MohammadReza0GH** [reply]
thanks a million, XD problem solved sir.
